testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -1387,24 +1386,17 @@ public void testAcidUpdateWithSubqueryPredicate()
                 onTrino().executeQuery(format("UPDATE %s SET column2 = 'another row updated' WHERE column1 = (SELECT min(regionkey) + 2 FROM %s)", tableName, secondTable));
                 verifySelectForTrinoAndHive("SELECT * FROM " + tableName, "true", row(1, "row updated"), row(2, "another row updated"));
             });
-
-            // WHERE with correlated subquery
-            assertThat(() -> onTrino().executeQuery(format("UPDATE %s SET column2 = 'row updated yet again' WHERE column2 = (SELECT name FROM tpch.tiny.region WHERE regionkey = column1)", tableName)))
-                    // TODO (https://github.com/trinodb/trino/issues/3325) support correlated UPDATE
-                    .failsWithMessageMatching("\\Qjava.sql.SQLException: Query failed (#\\E\\S+\\Q): Invalid descendant for DeleteNode or UpdateNode: io.trino.sql.planner.plan.MarkDistinctNode");
```

Comment:
Why removed?
Please keep this in, just checking the expected result from query failure to success, and add post-UPDATE result verification with `verifySelectForTrinoAndHive`

Reply:
bump

Reply:
But won't we use `assertThat` for queries which are failing ?

Reply:
You're correct. i was imprecise about "keep this in". here, "this" supposed to mean this test query, not the whole assertion

Reply:
But the above query will run successfully but it won't update the data. Is that okay ? We might need to add another query that updates the actual row

Reply:
> But the above query will run successfully but it won't update the data. Is that okay ?

indeed. The point of this was a query that's a bit non trivial, hence column1, and column2 are involved.

```
// WHERE with correlated subquery
onTrino().executeQuery(format("INSERT INTO %s VALUES (3, 'EUROPE')", tableName));
onTrino().executeQuery(format("UPDATE %s SET column2 = 'row updated yet again' WHERE column2 = (SELECT name FROM tpch.tiny.region WHERE regionkey = column1)", tableName));
verifySelectForTrinoAndHive("SELECT * FROM " + tableName, "true", row(1, "row updated"), row(2, "another row updated"), row(3, "row updated yet again"));
```

Reply:
Have applied it.